pkg/sharings/sharings.go

```diff
@@ -204,8 +202,39 @@ func ShareDoc(instance *instance.Instance, sharing *Sharing, recStatus *Recipien
 			}
 		}
 
+		var values []string
+
+		// Dynamic sharing
+		if rule.Selector != "" {
+			// Create index based on selector to retrieve documents to share
+			index := mango.IndexOnFields(docType, sharing.ID(), []string{rule.Selector})
```

Comment:
Index creation is idempotent, it does not return an error if the index already exists.